Marlin/src/HAL/STM32/HAL.cpp

```diff
@@ -48,8 +48,8 @@ uint16_t HAL_adc_result;
 // Public functions
 // ------------------------
 
-// Needed for DELAY_NS() / DELAY_US() on CORTEX-M7
-#if (defined(__arm__) || defined(__thumb__)) && __CORTEX_M == 7
+// Needed for DELAY_NS() / DELAY_US() on CORTEX-M4 to M7
+#if (defined(__arm__) || defined(__thumb__)) && WITHIN(__CORTEX_M, 4, 7) && !defined(TARGET_LPC1768)
```

Comment:
I think you can remove LPC. It's M3 and already excluded by previous check

Reply:
Probably this "constructor" code may be moved to shared

Reply:
Ok. I didn't test it. I was just following that old issue. I will test.